keps/sig-node/4622-topologymanager-max-allowable-numa-nodes/README.md

```diff
@@ -0,0 +1,757 @@
+
+what's the maxAllowableNUMANodes? maxAllowableNUMANodes specifies the maximum number of NUMA Nodes that the TopologyManager supports on the underlying machine. At present, having more than this number of NUMA Nodes will result in a state explosion when trying to enumerate possible NUMAAffinity masks and generate hints for them. As such, if more NUMA Nodes than this are present on a machine and the TopologyManager is enabled, an error will be returned and the TopologyManager will not be loaded.
+
+to configure the value of the maximum number of NUMA cells (or, equivalently, "NUMA nodes") the Topology Manager will take into account when computing the optimal allocation. Because of the algorithm being used in the Topology Manager, we need a hard cap to the number of NUMA cells being considered. At present day, having more than this number of NUMA cells will result in a state explosion when trying to enumerate possible NUMA affinity masks and generate hints for them.
+The current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager behavior and support modern high-end CPUs which are more and more likely to expose more than 8 NUMA cells.
+
+## Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+- Support high-end CPUs with more than 8 NUMA cells.
+
+## Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+- This proposal does not aim to modify the existing TopologyManager Policies. It focuses solely on introducing a new policy option to let users configure the maximum supported number of NUMA nodes.
+- It does not address other resource allocation or management aspects within Kubernetes.
+- Improve the topology manager to remove the state explosion.
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+We propose to add a new `TopologyManager` policy option called `max-allowable-numa-nodes` to the static TopologyManager policy. It can configure the value of maxAllowableNUMANodes in the TopologyManager. the current hard-coded value of 8 was added as a stop-gap 4 years ago, a configurable option policy option can improve the topology manager to remove the state explosion that occurs when trying to enumerate the possible NUMA affinities and generating their hints.
+
```

Comment:
I think we can just drop this section. The change is so minimal that the intro says it all.

Reply:
Yeah, dropped.